base/x509_utils.go

```diff
@@ -0,0 +1,42 @@
+package base
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"sync/atomic"
+)
+
+// loadCertificatesFromFile returns a new CertPool with all the PEM-format certificates in the given file.
+func loadCertificatesFromFile(path string) (*x509.CertPool, error) {
+	pool := x509.NewCertPool()
+	certs, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load CA certificate from %s: %w", MD(path).Redact(), err)
+	}
+	ok := pool.AppendCertsFromPEM(certs)
+	if !ok {
+		return pool, fmt.Errorf("did not load any valid certificate from file %s: %w", MD(path).Redact(), err)
+	}
+	return pool, nil
+}
+
+// systemCertPool is used to lazy-load the system cert pool in getSystemCertPool
+var systemCertPool atomic.Value
+
+// getSystemCertPool returns a CertPool for the system root store.
+func getSystemCertPool() *x509.CertPool {
+	val := systemCertPool.Load()
+	if val != nil {
+		return val.(*x509.CertPool)
+	}
+	pool, err := x509.SystemCertPool()
```

Comment:
SystemCertPool support for Windows was only added in 1.18, which I think means we may have to handle this slightly more gracefully?

In the other x509 util, we create an empty cert pool and let it fail when connecting if we error.